### DISCUSSION TITLE
Snapshot: override_status_bar not working

### DISCUSSION THREAD

**Discussion by simonnickel** [fastlane/fastlane#17021] [Q&A]
Hey, I am just creating some screenshots and noticed the override_status_bar(true) does not work.

Log output says it is applied: 

```
[17:20:31]: Overriding Status Bar
[17:20:31]: $ xcrun simctl status_bar C7AEB8B9-91C6-497B-ABEB-69CECA9C9C1D override --time 2007-01-09T09:41:00+01:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100 &> /dev/null
```

Not sure if this is an Simulator tooling issue, using Xcode Version 11.6 (11E708), Simulator Version 11.6 (921.9.1)

Might it be related to my configuration? Is anyone experiencing the same?

**Comment by joshdholtz** [comment]
@simonnickel Hey! Are you able to test on an older version of Xcode and/or simulator by any chance? 😊 There is also a new option for `headless` that may help? The sims would usually run in the background but I think there were some recently reports of things not working great for status bar (or something like that) when running in background. So maybe try running with `headless: false` and see if that works? 🤷‍♂️

**Reply by simonnickel** [reply]
Tried with headless, did not change anything.
Also tried to disable concurrent_simulators and erase_simulator.

Using the following config: 
```
concurrent_simulators true
clear_previous_screenshots true
override_status_bar true
stop_after_first_error true
skip_open_summary true
clean true
reinstall_app true
erase_simulator true
headless false
```

**Reply by joshdholtz** [reply]
Have you ran with `--verbose` to see if the `xcrun simctl` commands are getting executed? 🤔

**Reply by NachoSoto** [reply]
This is also not working for me. I'm running with `verbose`, and the command is being run:
```
xcrun simctl status_bar A13A3A37-B09B-42BF-8579-108DF9060A19 override --time 2007-01-09T09:41:00-08:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100 &> /dev/null
```
However, I figured the `&> /dev/null` was hiding some error, so I ran it myself... and it exits with no errors and code 0. However, the simulator status bar is unchanged. So I wonder if this is an Xcode regression? It hasn't been working for a few months for me.

**Reply by joshdholtz** [reply]
@NachoSoto Heyyyyy 👋  I think I've heard other people having these errors and it just not working when run manually either. I know I've had a conversation about this somewhere with someone.  But I can try to take a look again 😊

**Reply by joshdholtz** [reply]
@simonnickel @NachoSoto I think https://github.com/fastlane/fastlane/pull/19344 fixes this! I tried that arguments you both were using and it didn't work for me (like it didn't work for you) on the latest release. I then tried them on this PR and they worked for me.

Sooooo... I'm hoping this is fixed for you both in next release of _fastlane_ should should be sometime between today and Friday 😊

**Comment by NachoSoto** [comment]
Tried with `2.194.0` and it's still not working. I noticed running with `--verbose` that the command (`xcrun simctl status_bar AB6B5DF2-D310-453B-9D4F-8E6011C8E69D override --time 2007-01-09T09:41:00-08:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100 &> /dev/null`) was ran way before the simulator had completely booted.

I'm running with all these options fyi:
```
concurrent_simulators false
reinstall_app false
erase_simulator false
clear_previous_screenshots false
override_status_bar true
localize_simulator true
number_of_retries 1
headless false
test_without_building true
```

**Comment by NachoSoto** [comment]
I copied the command and ran it while the test is running to see the output:
```
xcrun simctl status_bar 5613561D-D6E5-4F93-BED9-5ECDB0A93B0E override --time 2007-01-09T09:41:00-08:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100
```
```
An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=405):
Unable to lookup in current state: Shutdown
```

**Comment by NachoSoto** [comment]
Hmmm I'm running it again and now it's working. I wonder if there's still some race condition?

**Comment by NachoSoto** [comment]
I'm running `xcrun simctl list | grep 'Booted'` and the simulator where the test is running isn't listed WTF

**Comment by NachoSoto** [comment]
Looks like the simulator gets set up correctly with the override status bar, then when the tests actually start, Xcode closes that sim, and opens a "Clone 1 of..." of that simulator, without the overriden status bar.

**Comment by Durbin7411** [comment]
> Hey, I am just creating some screenshots and noticed the override_status_bar(true) does not work.
> 
> Log output says it is applied:
> 
> ```
> [17:20:31]: Overriding Status Bar [UPSers](https://www.upsers.onl/)
> [17:20:31]: $ xcrun simctl status_bar C7AEB8B9-91C6-497B-ABEB-69CECA9C9C1D override --time 2007-01-09T09:41:00+01:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100 &> /dev/null
> ```
> 
> Not sure if this is an Simulator tooling issue, using Xcode Version 11.6 (11E708), Simulator Version 11.6 (921.9.1)
> 
> Might it be related to my configuration? Is anyone experiencing the same?

Did you have a fix on this issue? Facing same issue but no response from anyone.

**Comment by SpicaThor** [comment]
I've run into that and upon trying to run the command directly got following:
```
~ xcrun simctl status_bar FA8D96A5-B0FB-43B5-922F-05DBEF6770EA override --time 2007-01-09T09:41:00+02:00 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --operatorName '' --cellularBars 4 --batteryState charged --batteryLevel 100
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=22):
Simulator device failed to complete the requested operation.
Invalid argument
Underlying error (domain=NSPOSIXErrorDomain, code=22):
	Invalid, non-ISO date/time string
	Invalid argument
```
